components/nexusoperations/link_converter.go

```diff
@@ -0,0 +1,130 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package nexusoperations
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/nexus-rpc/sdk-go/nexus"
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/api/enums/v1"
+)
+
+func ConvertLinkWorkflowEventToNexusLink(we *commonpb.Link_WorkflowEvent) (nexus.Link, error) {
+	u, err := url.Parse(fmt.Sprintf(
+		"temporal:///namespaces/%s/workflows/%s/%s/history",
+		we.GetNamespace(),
+		we.GetWorkflowId(),
+		we.GetRunId(),
+	))
+	if err != nil {
+		return nexus.Link{}, err
+	}
+
+	linkType := ""
+	switch ref := we.GetReference().(type) {
+	case *commonpb.Link_WorkflowEvent_EventRef:
+		u.RawQuery = convertLinkWorkflowEventEventReferenceToURLQuery(ref.EventRef)
+		linkType = string(ref.EventRef.ProtoReflect().Descriptor().FullName())
+	}
+	return nexus.Link{URL: u, Type: linkType}, nil
+}
+
+func ConvertNexusLinkToLinkWorkflowEvent(link nexus.Link) (*commonpb.Link_WorkflowEvent, error) {
+	we := &commonpb.Link_WorkflowEvent{}
+	if !strings.HasPrefix(link.Type, string(we.ProtoReflect().Descriptor().FullName())+".") {
```

Comment:
Why the extra `.`?

Reply:
Just in case someone defines another type with this one as prefix.

Reply:
Not following... we only support the exact fullname right now, you can check equality instead of prefix match.

Reply:
So, because we also event reference, we need to add the event reference type to there. So, it's actual something like this: `temporal.api.common.v1.Link.WorkflowEvent.EventReference`

Reply:
The type should be `temporal.api.common.v1.Link.WorkflowEvent` and we should be able to infer the reference type from the query params. Maybe add a query param that says it's an event reference?